### PR TITLE
MAVLab course2017 direct control airframe

### PR DESCRIPTION
**Pull request by dewagter**

### CODE REVIEW

**Comment by kirkscheper** [comment]
@tomvand can you take a look at this at the next paparazzi day?

**Comment by tomvand** [comment]
Sure, will do

**Comment by OpenUAS** [comment]
Re-inited and Tom will fix it :)

**Comment by tomvand** [comment]
I could not finish this PR today, but I did do a first test flight using the aircraft in the course2018 conf. Some notes:
- The controller does not work, sometimes the drone does not respond to position errors or joystick input. This behavior changes throughout the flight. This might be caused by conflicting commands from the 'Standby' block in the flight plan. The flight plan does not seem to work well with module mode in general.
- The grass detector seems to work ok. The drone responded to the blue and black sections of the cyberzoo floor.
- Landings are rough, the drone cuts the throttle at ~0.5m
- The 4.5m geofence in the flight plan appears to cut the throttle, is this the intended behavior?
- The control loop behaves poorly, it really needs a damping term.
- The control module still needs the drone's position and heading to calculate attitude setpoints. This needs to be replaced before this airframe can be used without optitrack. The magnetometer did not work that well in the cyberzoo.

**Comment by OpenUAS** [comment]
Point 4, surly not, a throtte level below nominal and into the block landhere

**Comment by tomvand** [comment]
Fixed point 4. Note that there is no automatic landing in attitude mode now.

**Comment by gautierhattenberger** [comment]
The CI builds are failing at the moment since in current master the gazebo_models submodule is tagged "shallow" but a commit have been made some hours ago and the latest doesn't match the one that is checked-out.
This might be solved with #2327 among other things

File: .gitmodules [hunk 1]
```diff
@@ -41,4 +41,6 @@
 [submodule "sw/ext/tudelft_gazebo_models"]
 	path = sw/ext/tudelft_gazebo_models
 	url = https://github.com/tudelft/gazebo_models.git
-	shallow = true
+[submodule "sw/ext/eigen"]
```
Comment (gautierhattenberger) [inline .gitmodules]: eigen is already in master now, this PR should probably be rebased

**Comment by gautierhattenberger** [comment]
if you want this for next master, can someone fix the remaining issues ?

**Comment by OpenUAS** [comment]
Help to fix validate an close PR is requested today in MAVLab...

**Comment by dewagter** [comment]
@tomvand : can we close this one? Everything will be overwritten by 2019, right?

**Comment by tomvand** [comment]
Yes, the 2019 course contains a working example with guidance commands. I'll probably make a pull request for that next week.

**Comment by dewagter** [comment]
Current version: https://github.com/tudelft/paparazzi/tree/mavlabCourse2019

**Pull request closed without merging**